websocketOutportDriver/sender/acknowledgesHolder.go

```diff
@@ -0,0 +1,45 @@
+package sender
+
+import "sync"
+
+type acknowledgesHolder struct {
+	acknowledges map[string]*websocketClientAcknowledgesHolder
+	mut          sync.Mutex
+}
+
+// NewAcknowledgesHolder returns a new instance of acknowledgesHolder
+func NewAcknowledgesHolder() *acknowledgesHolder {
+	return &acknowledgesHolder{
+		acknowledges: make(map[string]*websocketClientAcknowledgesHolder),
+	}
+}
+
+// AddEntryForAddress will add the client to the inner map
+func (ah *acknowledgesHolder) AddEntryForAddress(remoteAddr string) {
+	ah.mut.Lock()
+	ah.acknowledges[remoteAddr] = NewWebsocketClientAcknowledgesHolder()
+	ah.mut.Unlock()
+}
+
+// GetAcknowledgesOfAddress will return the acknowledges for the specified address, if any
+func (ah *acknowledgesHolder) GetAcknowledgesOfAddress(remoteAddr string) (*websocketClientAcknowledgesHolder, bool) {
```

Comment:
if we ever want to go further, this can be made to return an interface that the `websocketClientAcknowledgesHolder`
In this way we can create an interface for the `acknowledgesHolder` implementation.

Reply:
no need for now as these components are very simple and we shouldn't mock them in the upcoming tests